test/e2e/prometheus_test.go

```diff
@@ -94,152 +90,175 @@ func TestAntiAffinity(t *testing.T) {
-	if err != nil {
-		t.Fatal(err)
-	}
+	prometheusReceiverURL := svc.Name + "." + svc.Namespace + ".svc.cluster.local"
 
 	// setup a self-signed ca and store the artifacts in a secret
+	secName := fmt.Sprintf("selfsigned-%s-bundle", name)
 	tlsSecret := &v1.Secret{
 		ObjectMeta: metav1.ObjectMeta{
-			Name:      "selfsigned-mtls-bundle",
+			Name:      secName,
 			Namespace: f.Ns,
 			Labels: map[string]string{
 				"group":                    name,
 				framework.E2eTestLabelName: framework.E2eTestLabelValue,
 			},
 		},
 		Data: map[string][]byte{
-			"client-cert-name": []byte("test-client"),
-			"serving-cert-url": []byte(deployedService.Spec.ClusterIP),
+			"client-cert-name": []byte("remoteWrite-client"),
+			"serving-cert-url": []byte(prometheusReceiverURL),
 		},
 	}
 	if err := createSelfSignedMTLSArtifacts(tlsSecret); err != nil {
 		t.Fatal(err)
 	}
-	if err := f.OperatorClient.CreateOrUpdateSecret(ctx, tlsSecret); err != nil {
+	if err := f.OperatorClient.CreateIfNotExistSecret(ctx, tlsSecret); err != nil {
 		t.Fatal(err)
 	}
 
 	// deploy remote write target
-	targetDeployment := fmt.Sprintf(`apiVersion: apps/v1
-kind: Deployment
-metadata:
-  name: instrumented-sample-app
-  namespace: openshift-monitoring
-  labels:
-    group: %[1]s
-    %[2]s
-spec:
-  replicas: 1
-  selector:
-    matchLabels:
-      group: %[1]s
-  template:
-    metadata:
-      labels:
-        group: %[1]s
-    spec:
-      containers:
-      - name: example-app
-        args:
-        - --cert-path=/etc/certs
-        image: quay.io/coreos/instrumented-sample-app:0.2.0-bearer-mtls-1
-        imagePullPolicy: IfNotPresent
-        ports:
-        - name: web
-          containerPort: 8080
-        - name: mtls
-          containerPort: 8081
-        volumeMounts:
-        - mountPath: /etc/certs
-          name: certs
-      volumes:
-      - name: certs
-        secret:
-          secretName: selfsigned-mtls-bundle
-          items:
-          - key: server-ca.pem
-            path: cert.pem
-          - key: server.key
-            path: key.pem
-`, name, framework.E2eTestLabel)
-	rwTestDeployment, err := manifests.NewDeployment(bytes.NewReader([]byte(targetDeployment)))
-	if err != nil {
+	prometheusReceiver := f.MakeBasicPrometheus(f.Ns, name, name, 1, []string{secName})
+	f.AddRemoteReceiveWithWebTLSToPrometheus(prometheusReceiver, tlsSecret)
+	if err := f.OperatorClient.CreateOrUpdatePrometheus(ctx, prometheusReceiver); err != nil {
 		t.Fatal(err)
 	}
-	if err := f.OperatorClient.CreateOrUpdateDeployment(ctx, rwTestDeployment); err != nil {
+	if err := f.OperatorClient.WaitForPrometheus(ctx, prometheusReceiver); err != nil {
+		t.Fatal(err)
+	}
+
+	route := f.MakePrometheusServiceRoute(svc)
+	if err := f.OperatorClient.CreateRouteIfNotExists(ctx, route); err != nil {
 		t.Fatal(err)
 	}
-	for _, scenario := range []struct {
-		name   string
-		port   string
-		rwSpec string
+
+	url, err := f.OperatorClient.GetRouteURL(ctx, route)
```

Comment:
Can't we use here the function `framework.NewPrometheusClientFromRoute`?

Reply:
Good point, I think that's a left-over from when I tried to get away without creating a `Route`.